### PR TITLE
Rebase Master branch with v1.8 features 

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind code-refactoring

**What does this PR do / why we need it**:
Rebase GitOps Operator Master branch with v1.8 branch changes. While it looks like there are many changes missing in the master branch, most of the them are the commits where I tried the render_templates implementation.

**How to test changes / Special notes to the reviewer**:
1. You can test this changes locally by running `make install run` and verify if the operator installation is successful. On successful installation operator creates all the Argo CD workloads in `openshift-gitops` namespace.

(or)

2. If you have an OpenShift cluster, you can also test this PR using OLM

```

make docker-build IMG=quay.io/aveerama/gitops-backend-operator:testiib
make docker-push IMG=quay.io/aveerama/gitops-backend-operator:testiib
rm -fr bundle
make bundle IMG=quay.io/aveerama/gitops-backend-operator:testiib
make bundle-build BUNDLE_IMG=quay.io/aveerama/gitops-backend-operator-bundle:testiib
make bundle-push BUNDLE_IMG=quay.io/aveerama/gitops-backend-operator-bundle:testiib
make catalog-build BUNDLE_IMG=quay.io/aveerama/gitops-backend-operator-bundle:testiib CATALOG_IMG=quay.io/aveerama/gitops-backend-operator-index:testiib
make catalog-push CATALOG_IMG=quay.io/aveerama/gitops-backend-operator-index:testiib
```

The above commands will create a Index Image. You can create a catalog source using the index Image and deploy it on to your OpenShift Cluster.

```
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: gitops-catalog
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/<>/gitops-backend-operator-index@<>
  displayName: GitOps Operators
  publisher: GitOps Community
```

### CODE REVIEW

File: bundle/manifests/gitops-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -596,30 +596,13 @@ spec:
...
-                ports:
-                - containerPort: 8443
-                  name: https
-                resources: {}
-              - args:
-                - --health-probe-bind-address=:8081
-                - --metrics-bind-address=127.0.0.1:8080
-                - --leader-elect
-                command:
-                - /manager
```
Comment (jaideepr97) [inline bundle/manifests/gitops-operator.clusterserviceversion.yaml:615]: @iam-veeramalla is this removal intentional?

Comment (iam-veeramalla) [reply bundle/manifests/gitops-operator.clusterserviceversion.yaml:615]: Yes Jaideep, I am trying to keep the CSV similar to the downstream one and I realised in the downstream we don't use this. 

If you build the bundle for this operator locally, it create the operator pod with 2 containers .. a proxy container which we don't use and the manager container. 

We don't use the proxy container in the downstream.. so just to keep them aligned I removed this.

**Review by jaideepr97 — DISMISSED** [review]
thanks @iam-veeramalla
